src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
That's true. Actually, my main objection against `Optional` is that it's not supposed to be used as field type or parameter type. And I don't truly understand why because current approach creates inconsistencies like the one above. If only Java had Kotlin's null-handling...